src/rgw/rgw_notify.cc

```diff
@@ -703,13 +707,11 @@ static inline void populate_event(reservation_t& res,
   set_event_id(event.id, etag, ts);
   event.bucket_id = res.bucket->get_bucket_id();
   // pass meta data
-  if (res.x_meta_map.empty()) {
-    // no metadata cached:
+  if (!res.metadata_fetched_from_attributes) {
     // either no metadata exist or no metadata filter was used
     metadata_from_attributes(res, obj);
-  } else {
-      event.x_meta_map = res.x_meta_map;
   }
+  event.x_meta_map = res.x_meta_map;
```

Comment:
this seems to be the heart of the fix? assignment of event.x_meta_map is now unconditional

Reply:
yes. this and fetching the attributes based on the flag instead of whether there are attributes (there could be attributes in an object update request, but we still need to fetch the original attributes if they were never cached.